pkg/controllers/workgenerator/controller_integration_test.go

```diff
@@ -1210,6 +1210,7 @@ var _ = Describe("Test Work Generator Controller", func() {
 					return errors.IsNotFound(err)
 				}, duration, interval).Should(BeTrue(), "controller should remove work in hub cluster that is no longer needed")
 				By(fmt.Sprintf("second work %s is deleted in %s", work.Name, work.Namespace))
+				verifyBindingStatusSyncedNotApplied(binding, false, false)
```

Comment:
in both case, eventually it should be applied true. if it is stuck, applied should be false.